### ISSUE TITLE
Rule Change: ignore array items destructuring the same way as  args: after-used

### ISSUE THREAD

**Issue by CostachescuCristinel** [eslint/eslint#15139]
### What rule do you want to change?

no-unused-vars

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// In this case we are only interested in the capturing groups matches
// fullMatch is not useful
const [fullMatch, cg1, cg2] = "Hello, eslint!".match(/(\w+)\W+(\w+)/gi);

const data = [
    ["id_1", "Soap", 4.49, 0, "Great for everyday use!"],
    ["id_2", "Shampoo", 12.99, 5, "Helps keep your hair strong!"],
    ["id_3", "Hand cleaner", 3.79, 0, "Delicate hand cleaner for delicate hands"]
];

data.reduce((result, productData) => {
    // In this situation, we're only interested into creating a shopping
    // list summary, and calculating the total amount
    // The id field is not useful for the purpose, but is a positional
    // destructuring variable and it must be included to properly
    // destructure the data fields
    // description is truly unused and not necessary, the rule should
    // report this variable
    const [id, name, price, discount, description] = productData;
    const finalPrice =  price - (price * (discount/100));
    
    result.products.push(`${name}: ${finalPrice.toFixed(2)}`);
    result.total += finalPrice;

    return result;
}, { products: [], total: 0 });
```

### What does the rule currently do for this code?

The `no-unused-vars` rule will report all unused variables from any `Array` destructuring code, without taking into account that they are a positional requirement - there is no way around destructuring an array otherwise that would not create such unused variables.

### What will the rule do after it's changed?

By adding a configuration option `ignoreArrayDestructuring`, the rule should behave according to the specified config, as follows:
- `none`: (default) All unused variables from an array destructuring will be reported.  
This option maintains the same behavior as the current rule implementation, and should be the default value.
- `after-used`: Behaves the same as `"args": "after-used"`.  
Reports as unused all variables destructured **after the last used** one.  
If no variable is used, all variables should be reported.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

I have never looked at eslint rules' code, and I don't think i would be able to properly implement such behavior. Someone with more experience than me should take on the task of implementing it.

It should be noted that the following JS syntax is legal in some of the latest ECMAScript specifications:
```
const [, name,, price,, discount, ] = ["id_1", "Soap", "something", 4.49, "something-else", 0, "Great for everyday use!"];
console.log(name, price, discount);
```

However, most developers (and I) agree that this syntax is not very readable. Multiple consecutive commas are hard to understand while looking at the code, if you're not paying careful attention. Also, in the example above, the first comma will most likely go unnoticed in the context of a few tens of lines of code.

**Comment by nzakas** [comment]
Thanks for the suggestion. I think you did stumbled upon what we recommend: you don’t need to define variables for destructing, you can just use commas without identifiers between them. This is different than with arguments where you must supply identifiers, so I don’t think the two situations are the same.

You can always use a disable comment in any specific instance where you don’t want to have this warning show up
